jib-core/src/test/java/com/google/cloud/tools/jib/builder/BuildConfigurationTest.java

```diff
@@ -200,6 +202,69 @@ public void testBuilder_missingValues() {
     }
   }
 
+  @Test
+  @SuppressWarnings("JdkObsolete")
+  public void testBuilder_nullValues() {
+    // Java arguments element should not be null.
+    try {
+      BuildConfiguration.builder(Mockito.mock(BuildLogger.class))
+          .setJavaArguments(Arrays.asList("first", null));
+      Assert.fail("The IllegalArgumentException should be thrown.");
+    } catch (IllegalArgumentException ex) {
+      Assert.assertNull(ex.getMessage());
+    }
+
+    // JVM flags element should not be null.
+    try {
+      BuildConfiguration.builder(Mockito.mock(BuildLogger.class))
+          .setJvmFlags(Arrays.asList("first", null));
+      Assert.fail("The IllegalArgumentException should be thrown.");
+    } catch (IllegalArgumentException ex) {
+      Assert.assertNull(ex.getMessage());
+    }
+
+    // Exposed ports element should not be null.
+    try {
+      BuildConfiguration.builder(Mockito.mock(BuildLogger.class))
+          .setExposedPorts(Arrays.asList(new Port(1000, Protocol.TCP), null));
+      Assert.fail("The IllegalArgumentException should be thrown.");
+    } catch (IllegalArgumentException ex) {
+      Assert.assertNull(ex.getMessage());
+    }
+
+    // Can accept empty environment
+    BuildConfiguration.builder(Mockito.mock(BuildLogger.class)).setEnvironment(ImmutableMap.of());
+
+    // Environment keys element should not be null.
+    Map<String, String> environmentMap = Maps.newHashMap();
+    environmentMap.put(null, "value");
+
+    try {
+      BuildConfiguration.builder(Mockito.mock(BuildLogger.class)).setEnvironment(environmentMap);
+      Assert.fail("The IllegalArgumentException should be thrown.");
+    } catch (IllegalArgumentException ex) {
+      Assert.assertNull(ex.getMessage());
+    }
+
+    // Environment values element should not be null.
+    environmentMap = Maps.newHashMap();
+    environmentMap.put("key", null);
+    try {
+      BuildConfiguration.builder(Mockito.mock(BuildLogger.class)).setEnvironment(environmentMap);
+      Assert.fail("The IllegalArgumentException should be thrown.");
+    } catch (IllegalArgumentException ex) {
+      Assert.assertNull(ex.getMessage());
+    }
+
+    // Environment map can accept TreeMap and Hashtable
+    try {
+      BuildConfiguration.builder(Mockito.mock(BuildLogger.class)).setEnvironment(Maps.newTreeMap());
+      BuildConfiguration.builder(Mockito.mock(BuildLogger.class)).setEnvironment(new Hashtable<>());
+    } catch (NullPointerException ex) {
```

Comment:
I see above you did not catch the exception.
```
    // Can accept empty environment
    BuildConfiguration.builder(Mockito.mock(BuildLogger.class)).setEnvironment(ImmutableMap.of());
```
Let's be consistent and remove this try-catch block, throwing potential NPEs to the test framework. (Moreover, if you were to catch, you should wrap individual calls to `setEnvironment()`.)

BTW, you'd normally do `Assert.fail()` instead of directly doing `throw new AssertionError()`.

Reply:
Yup, let me revise the test, however, for `Assert.fail()` I did that and it violate the error-prone checking so that why I change to `throw new AssertionError()`